src/_pytest/_code/code.py

```diff
@@ -662,10 +662,11 @@ def match(self, regexp: Union[str, Pattern[str]]) -> "Literal[True]":
         If it matches `True` is returned, otherwise an `AssertionError` is raised.
         """
         __tracebackhide__ = True
-        msg = "Regex pattern {!r} does not match {!r}."
-        if regexp == str(self.value):
-            msg += " Did you mean to `re.escape()` the regex?"
-        assert re.search(regexp, str(self.value)), msg.format(regexp, str(self.value))
+        value = str(self.value)
+        msg = f"Regex pattern did not match.\n Regex: {regexp!r}\n Input: {value!r}"
```

Comment:
I wonder if it wouldn't be better to use a straight str here instead of repr. With the new line format the quoting doesn't seem needed, perhaps the opposite.

(Either way is fine with me)

Reply:
i believe for the regex it helps to get the escapes printed more nicely readable, my personal opinion is that repr reads slightly better than just the str or the escape

Reply:
i just experimented with this  - if there is a simple way to print "raw" strings, i'll happily use that - however even in the pytest testsuite there is a number of cases that would turn confusing without

Reply:
> if there is a simple way to print "raw" strings, i'll happily use that

That would be `{regexp}` instead of `{regexp!r}`, unless I'm misunderstanding.

> however even in the pytest testsuite there is a number of cases that would turn confusing without

Can you give an example?

Reply:
Hmm, I'm somewhat undecided here. I agree the double backslashes are a bit confusing, but at the same time I think it's good to print the repr for the input (considering that it can contain whitespace and such, and seeing that whitespace might be essential to find out why the pattern doesn't match).

But then if we use the repr for the input, it might be more consistent and thus less confusing if we use the repr for the pattern as well.